test/module/irohad/ametsuchi/tx_presence_cache_test.cpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "ametsuchi/impl/tx_presence_cache_impl.hpp"
+#include "cryptography/public_key.hpp"
+#include "interfaces/common_objects/transaction_sequence_common.hpp"
+#include "interfaces/iroha_internal/transaction_batch_factory_impl.hpp"
+#include "interfaces/iroha_internal/transaction_batch_impl.hpp"
+#include "module/irohad/ametsuchi/ametsuchi_mocks.hpp"
+#include "module/shared_model/interface/mock_transaction_batch_factory.hpp"
+#include "module/shared_model/interface_mocks.hpp"
+
+using namespace iroha::ametsuchi;
+using namespace testing;
+
+class TxPresenceCacheTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    mock_storage = std::make_shared<MockStorage>();
+    mock_block_query = std::make_shared<MockBlockQuery>();
+    EXPECT_CALL(*mock_storage, getBlockQuery())
+        .WillRepeatedly(Return(mock_block_query));
+  }
+
+ public:
+  std::shared_ptr<MockStorage> mock_storage;
+  std::shared_ptr<MockBlockQuery> mock_block_query;
+};
+
+/**
+ * @given hash which is not in storage
+ * @when cache asked for hash status
+ * @then cache returns Missing status
+ */
+TEST_F(TxPresenceCacheTest, MissingHashTest) {
```

Comment:
This and two following tests are the same. You can make them type-parameterized (https://github.com/abseil/googletest/blob/master/googletest/docs/advanced.md#type-parameterized-tests). You can check the example in permissions_test.cpp